datashape/discovery.py

```diff
@@ -405,3 +405,19 @@ def descendents(d, x):
         children -= desc
         desc.update(children)
     return desc
+
+
+Mock = None
+try:
+    from unittest.mock import Mock
+except ImportError:
+    try:
+        from mock import Mock
+    except ImportError:
+        pass
+
+if Mock is not None:
+    @dispatch(Mock)
```

Comment:
This will probably break some code somewhere. If we are okay with that then I agree that it is a better solution.